### PR TITLE
Backend functionality to create/edit an event

### PR DESCRIPTION
**Pull request by munteanumarco**
Add authorization attributes to controllers to ensure we let the right users access the right things with the right privileges.
I also added swagger support for  bearer token (better development experience and easier testability)
Modified also profile behaviour - now when a new user is created will also get an empty profile (so I got rid of the create endpoint for user profile as its no longer needed)

### CODE REVIEW

File: EventPlannerAPI/BusinessLayer/Interfaces/IUserService.cs [hunk 1]
```diff
@@ -15,7 +15,6 @@ public interface IUserService
 		Task<EventPlannerUser> GetUserByIdentifier(string userIdentifier);
         Task<GetUserProfileDetailsDto> GetUserProfileDetailsAsync(string userId);
         Task<IList<string>> GetRolesAsync(EventPlannerUser user);
-        Task<GetUserProfileDetailsDto> CreateUserProfileDetailsAsync(string userId, UpsertUserProfileDetailsDto userDetails);
```
Comment (HellHound97) [inline EventPlannerAPI/BusinessLayer/Interfaces/IUserService.cs:18]: Why did you delete these?

**Review by HellHound97 — APPROVED** [review]

**Review by gheorghe-sava — APPROVED** [review]
